osquery/tables/networking/linux/tests/namespace_tests.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <gtest/gtest.h>
+
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/tests/test_additional_util.h"
+#include "osquery/tests/test_util.h"
+
+#include <netinet/in.h>
+#include <sched.h>
+#include <signal.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+namespace {
+const int kNamespaceTestPort = 27960;
+
+// It is easier to allocate these structures here instead of inside the
+// new entry point (which may cause stack issues)
+struct sockaddr_in server_addr = {};
+struct sockaddr_in client_addr = {};
+
+// The new entry point used by the listening_ports_table test; this
+// code just listens for a new client and terminates after a connection
+// has been received
+int NetNamespaceTests_CloneEntryPoint(void*) {
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_addr.s_addr = htonl(INADDR_ANY);
+  server_addr.sin_port = htons(kNamespaceTestPort);
+
+  auto server_socket = socket(AF_INET, SOCK_STREAM, 0);
+  if (server_socket == -1) {
+    std::cerr << "Failed to create the socket\n";
```

Comment:
Why `std::cerr`? Why not make use of `LOG(ERROR)`?

Reply:
I don't expect this output to be show up or be important as this is just the secondary entry point and any possible error is caught by the main process. I'm mainly afraid to make it crash when calling those functions. I used std::cerr because I found out it was working fine